cli/internal/env/env.go

```diff
@@ -26,15 +26,50 @@ type DetailedMap struct {
 	BySource BySource
 }
 
-// Merge takes another EnvironmentVariableMap and merges it into the receiver
-// It overwrites values if they already exist, but since the source of both will be os.Environ()
-// it doesn't matter
-func (evm EnvironmentVariableMap) Merge(another EnvironmentVariableMap) {
+// EnvironmentVariablePairs is a list of "k=v" strings for env variables and their values
+type EnvironmentVariablePairs []string
+
+// WildcardMaps is a pair of EnvironmentVariableMaps.
+type WildcardMaps struct {
+	Inclusions EnvironmentVariableMap
+	Exclusions EnvironmentVariableMap
+}
+
+// Resolve collapses a WildcardSet into a single EnvironmentVariableMap.
+func (ws WildcardMaps) Resolve() EnvironmentVariableMap {
+	output := EnvironmentVariableMap{}
+	output.Union(ws.Inclusions)
+	output.Difference(ws.Exclusions)
+	return output
+}
+
+// GetEnvMap returns a map of env vars and their values from os.Environ
+func GetEnvMap() EnvironmentVariableMap {
+	envMap := make(map[string]string)
+	for _, envVar := range os.Environ() {
+		if i := strings.Index(envVar, "="); i >= 0 {
+			parts := strings.SplitN(envVar, "=", 2)
+			envMap[parts[0]] = strings.Join(parts[1:], "")
```

Comment:
This is a "break it you buy it" situation. I simply moved the function definition up in the file to make the file read more-reasonably top-to-bottom.

I'll improve the docs. The behavior is actually correct in case the value has a `=` in it.

Reply:
https://github.com/vercel/turbo/blob/main/cli/internal/env/env.go#L97-L107